src/docdb/nodes.ts

```diff
@@ -100,18 +106,43 @@ export class DocDBCollectionNode implements INode {
 	}
 
 	async getChildren(): Promise<INode[]> {
```

Comment:
You have too many functions (`getChildren`, `addChildrenFromDocuments`, `addMoreChildren` and `listDocuments`) that are too similar in functionality with too much copied code. Let's try to combine it into one or two functions. We can discuss offline if you need more details